### PR TITLE
lightning: fix int handle overflow and add retry for create schema

### LINKED ISSUE #1290: Lightning failed to import DDL "Error 9007: Write conflict"
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

- Edit lightning.toml :
image

- Import DDL: tiup tidb-lightning -config  lightning.toml
- Meanwhile, sysbench write to other database.
- After a few minutes, lightning failed with error message:
image

- 16GB memoryy tidb OOM after 4w+ tables has been written.

2. What did you expect to see?

- Lightning import all DDL&DML successfully.

3. What did you see instead?

- Fail to import data, Error 9007: Write conflict

4. What version of BR and TiDB/TiKV/PD are you using?

br -V
tidb-lightning -V
[release-version=v5.1.0] [git-hash=512e1ff98ea70fa45c4f2c27a690079b783bfefa] [git-branch=heads/refs/tags/v5.1.0] [go-version=go1.16.4] [utc-build-time="2021-06-17 07:20:57"] [race-enabled=false]

tidb-server -V
Release Version: v5.0.2
Edition: Community
Git Commit Hash: dc40a093a0058b95fc859f60d4b84337dbdff9f6
Git Branch: heads/refs/tags/v5.0.2
UTC Build Time: 2021-06-07 08:42:39
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false

tikv-server -V
TiKV
Release Version:   5.0.2
Edition:           Community
Git Commit Hash:   52f335601d4806b44454fb631a006d7fed5b4a50
Git Commit Branch: cloud-cdc-5.0
UTC Build Time:    2021-06-23 08:59:38
Rust Version:      rustc 1.51.0-nightly (bc39d4d9c 2021-01-15)
Enable Features:   jemalloc mem-profiling portable sse protobuf-codec test-engines-rocksdb cloud-aws cloud-gcp
Profile:           dist_release

pd-server -V
Release Version: v5.0.2
Edition: Community
Git Commit Hash: 2b5ada0d1fbfb559bb31cbba759df9793bd140d2
Git Branch: heads/refs/tags/v5.0.2
UTC Build Time:  2021-06-07 08:41:12

-->

5. Operation logs
   - Please upload `br.log` for BR if possible
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs

6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `topology.yml` if deployed by TiUP

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus if possible

### LINKED ISSUE #1291: Lightning ERROR "["startKey > endKey when paginating scan region"] [startKey=7480000000000307FFB85F720000000000FF0000000000000000FA] [endKey=7480000000000307FFB85F720000000000FF0000000000000000FA]"
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

- Edit lightning.toml:
 
image

-  Import data "tiup tidb-lightning -config  lightning.toml"
- KV region-split-size set to 20MB, region-max-size set to 30MB.
- Before lightning importing, some tables already existed in TiDB.

2. What did you expect to see?

3. What did you see instead?

When lightning importing,  some tables has error "["startKey > endKey when paginating scan region"
image

image

4. What version of BR and TiDB/TiKV/PD are you using?

br -V
tidb-lightning -V
5.1.0
tidb-server -V
5.1.0
tikv-server -V
5.1.0
pd-server -V
5.1.0
-->

5. Operation logs
   - Please upload `br.log` for BR if possible
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs

6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `topology.yml` if deployed by TiUP

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus if possible

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
- Fix the bug that lightning failed when int handle value `int64.MaxValue` . close #1291
- Add retry for `restoreSchema`. close #1290

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note

 - Fix the bug that int handle overflow and add retry for create schema

### CODE REVIEW

**Comment by kennytm** [comment]
```
[2021-06-28T07:14:27.056Z] FAIL: restore_test.go:1518: restoreSchemaSuite.TestRestoreSchemaSuccessful

[2021-06-28T07:14:27.056Z] 

[2021-06-28T07:14:27.056Z] /nfs/cache/mod/github.com/golang/mock@v1.4.4/gomock/controller.go:140:

[2021-06-28T07:14:27.056Z]     r.TestHelper.Fatalf(format, args...)

[2021-06-28T07:14:27.056Z] ... Error: Unexpected call to *mock.MockGlue.GetDB([]) at /home/jenkins/agent/workspace/br_ghpr_build/go/src/github.com/pingcap/br/pkg/mock/glue.go:74 because: there are no expected calls of the method "GetDB" for that receiver

[2021-06-28T07:14:27.056Z] 

[2021-06-28T07:14:27.056Z] restore_test.go:1520:

[2021-06-28T07:14:27.056Z]     c.Assert(err, IsNil)

[2021-06-28T07:14:27.056Z] ... value *errors.errorString = &errors.errorString{s:"context canceled"} ("context canceled")

[2021-06-28T07:14:27.056Z] 

[2021-06-28T07:14:27.056Z] OOPS: 63 passed, 1 FAILED
```

File: pkg/lightning/backend/local/local_test.go [hunk 1]
```diff
@@ -81,11 +81,17 @@ func (s *localSuite) TestNextKey(c *C) {
...
-	for _, handleID := range []int64{1, 255, math.MaxInt32} {
+	for _, handleID := range []int64{math.MinInt64, 1, 255, math.MaxInt32 - 1} {
 		key := tablecodec.EncodeRowKeyWithHandle(1, tidbkv.IntHandle(handleID))
 		c.Assert(nextKey(key), DeepEquals, []byte(tablecodec.EncodeRowKeyWithHandle(1, tidbkv.IntHandle(handleID+1))))
 	}
 
+	// overflowed
+	key := tablecodec.EncodeRowKeyWithHandle(1, tidbkv.IntHandle(math.MaxInt64))
+	next = tablecodec.EncodeTablePrefix(2)
+	c.Assert(bytes.Compare(key, next) < 0, IsTrue)
```
Comment (kennytm) [inline pkg/lightning/backend/local/local_test.go:92]:
```suggestion
	c.Assert(key, Less, next)
```

Comment (glorv) [reply pkg/lightning/backend/local/local_test.go:92]: The less check doesn't support custom type `kv.Key` 😅

**Review by kennytm — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**